pilot/pkg/security/authz/builder/builder.go

```diff
@@ -264,42 +299,62 @@ func (b Builder) buildHTTP(rule *builtRule, logger *AuthzLogger) []*hcm.HttpFilt
 		}
 	}
 
-	extauthz, err := getExtAuthz(b.extensions, providers)
-	if err != nil {
-		logger.AppendError(multierror.Prefix(err, "failed to process CUSTOM action, will generate deny configs for the specified rules:"))
-		rbac := &rbachttp.RBAC{Rules: getBadCustomDenyRules(rules)}
-		return []*hcm.HttpFilter{
-			{
+	// For CUSTOM policies, generate filter pairs for each unique provider
+	var filters []*hcm.HttpFilter
+
+	// Get unique providers and sort for deterministic output
+	uniqueProviders := maps.Keys(rule.providerRules)
+	sort.Strings(uniqueProviders)
+
+	for _, provider := range uniqueProviders {
```

Comment:
this changes the semantics of a failure significantly - if any one unique provider fails, there may or may not be an RBAC filter added for a previous provider, depending on the name of the provider. Currently, if provider translation fails, the rule will be denied (fail-closed). I think that it would make more sense to keep the current semantics, where failed translation generates deny rules for the target (workload/listener).